src/GeneralStateTestsFiller/stCreateTest/CREATE_EOF1Filler.yml

```diff
@@ -41,33 +41,239 @@ CREATE_EOF1:
       #- ':yul { mstore8(0, 0xef) return(0, 32) }'
       - ':label deploying_0xef00..00 :raw 0x60ef60005360206000f3'
 
-      # Valid EOF1 initcode and code combinations (both with and withoug data section)
+      # Valid EOF1 initcode and code combinations (both with and without data section)
 
       # Legacy initcode => EOF code
       # deployed code without data section
       # deployed code section: 'PUSH1 0 STOP'
       - ':label deploying_valid_eof1 :yul { mstore(0, 0xef00010100030060000000000000000000000000000000000000000000000000) return(0, 10) }'
-      # deployed code with data section
+      # deployed code section: 'PUSH1 0 PUSH1 0 RETURN'
+      - ':label deploying_valid_eof1_return :yul { mstore(0, 0xef00010100050060006000f30000000000000000000000000000000000000000) return(0, 12) }'
+      # deployed code section: 'PUSH1 01 PUSH1 00 SSTORE REVERT'
+      - ':label deploying_valid_eof1_revert :yul { mstore(0, 0xef0001010006006001600055fd00000000000000000000000000000000000000) return(0, 13) }'
+      # deployed code section: 'PUSH1 0 INVALID'
+      - ':label deploying_valid_eof1_invalid :yul { mstore(0, 0xef0001010003006000fe00000000000000000000000000000000000000000000) return(0, 10) }'
+      # deployed code section: 'PUSH1 00 SELFDESTRUCT'
+      - ':label deploying_valid_eof1_selfdestruct :yul { mstore(0, 0xef0001010003006000ff00000000000000000000000000000000000000000000) return(0, 10) }'
+
+     # depeoyed code with data section
       # deployed code section: 'PUSH1 0 STOP' data section: 'aabbccdd'
       - ':label deploying_valid_eof1_with_data :yul { mstore(0, 0xef000101000302000400600000aabbccdd000000000000000000000000000000) return(0, 17) }'
+      # deployed code section: 'PUSH1 0 PUSH1 0 RETURN' data section: 'aabbccdd'
+      - ':label deploying_valid_eof1_with_data_return :yul { mstore(0, 0xef00010100050200040060006000f3aabbccdd00000000000000000000000000) return(0, 19) }'
+      # deployed code section: 'PUSH1 01 PUSH1 00 SSTORE REVERT' data section: 'aabbccdd'
+      - ':label deploying_valid_eof1_with_data_revert :yul { mstore(0, 0xef0001010006020004006001600055fdaabbccdd000000000000000000000000) return(0, 20) }'
+      # deployed code section: 'PUSH1 0 INVALID' data section: 'aabbccdd'
+      - ':label deploying_valid_eof1_with_data_invalid :yul { mstore(0, 0xef0001010003020004006000feaabbccdd000000000000000000000000000000) return (0, 17) }'
+      # deployed code section: 'PUSH1 00 SELFDESTRUCT' data section: 'aabbccdd'
+      - ':label deploying_valid_eof1_with_data_selfdestruct :yul { mstore(0, 0xef0001010003020004006000ffaabbccdd000000000000000000000000000000) return(0, 17) }'
+
+      # EIP-3670: Legacy initcode unaffected by validation containing truncated PUSH data at the end returning valid EOF code
```

Comment:
Well this section is legacy cases, basically testing how legacy code behaves with truncated PUSHes... no change here expected, I think I would be fine here for simplicity leaving just a couple of cases, not testing all possible PUSHes

Reply:
Yes, I was hesitating to add this validation. I have removed most of them now, only left he ones for push1, push2, push31 and push32.